tools/small_rna_map/small_rna_map.r

```diff
@@ -0,0 +1,40 @@
+library(optparse)
+library(ggplot2)
+library(gridExtra)
+
+option_list <- list(
+    make_option(c("-r", "--output_tab"), type="character", help="path to tabular file"),
+    make_option("--output_pdf", type = "character", help="path to the pdf file with plot")
+    )
+
+parser <- OptionParser(usage = "%prog [options] file", option_list = option_list)
+args = parse_args(parser)
+
+theme_set(theme_bw())
+Table = read.delim(args$output_tab, header=T, row.names=NULL)
+
+for(i in 1:nrow(Table)){
+  if (Table$Polarity[i] == "R"){
+	Table$Nbr_reads[i]= -1 * Table$Nbr_reads[i]
+	}
+}
```

Comment:
R loops are very slow. Use R function whose "internal" looping is implemented in C:
lines 16 to 20 can be replaced by
Table <- within(Table, Nbr_reads[Polarity=="R"] <- (Nbr_reads[Polarity=="R"]*-1))